### PR TITLE
feat(issue): add `bug_report`, `feature_request` issue template

### PR DESCRIPTION
**Pull request by pysan3**
- Greatly inspired by: https://github.com/folke/noice.nvim/tree/main/.github/ISSUE_TEMPLATE

Came from the discussion https://github.com/nvim-neo-tree/neo-tree.nvim/discussions/1006.

## TODOs

- [ ] Please check if `# Your Configuration` section template is OK for you.

## Mentions

@miversen33 @cseickel @nhat-vo

### CODE REVIEW

**Comment by cseickel** [comment]
This is fine by me, but I'd like to hear @miversen33's thought since it was his idea.

**Comment by miversen33** [comment]
@pysan3 added a quick comment to the PR.

I tested the repro config on a windows machine just to verify that it works on there as well (lots of people in the neovim community assume linux so I wanted to cover our bases). It does indeed work on windows as well so that's awesome.

Thanks for getting this together!

**Comment by pysan3** [comment]
The config part is mostly done by folke and I just added `local neotree_config = ...` so big thanks to him ;)

**Comment by pysan3** [comment]
Anyways, thanks for your feedbacks and I think it's safe to merge this PR and have a way better debugging experience for us @cseickel !!

**Review by cseickel — APPROVED** [review]

**Comment by miversen33** [comment]
Lol so uhh, we [aren't making the repro configuration required](https://github.com/nvim-neo-tree/neo-tree.nvim/pull/1013#discussion_r1241247024)?

**Comment by cseickel** [comment]
> @pysan3 added a quick comment to the PR.

> Lol so uhh, we https://github.com/nvim-neo-tree/neo-tree.nvim/pull/1013#discussion_r1241247024?

Did you forget to submit a pending review?

**Comment by miversen33** [comment]
My comment is listed as pending. I wonder if since I'm not marked as required or something, GitHub didn't make my pending review prevent the merge? Idk, just spit balling

**Comment by cseickel** [comment]
Pending means "draft", only you can see it. You have to submit the pending review for anyone else to see your comments.

**Comment by miversen33** [comment]
> Pending means "draft", only you can see it. You have to submit the pending review for anyone else to see your comments.

Huh check that out. I thought simply commenting on the code snippet was good enough, but it seems that you have to push your review through. Good catch @cseickel :)

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        table.insert(plugins, neotree_config)
+        require("lazy").setup(plugins, {
+          root = root .. "/plugins",
+        })
+
+        vim.cmd.colorscheme("tokyonight")
+        -- add anything else here
+      render: Lua
+    validations:
+      required: false
```
[the remark this answers is not in the document]
Comment (cseickel) [reply .github/ISSUE_TEMPLATE/bug_report.yml:109]: I'll make this change right now.

**Comment by pysan3** [comment]
Lol I did the exact same thing just a few days ago: https://github.com/nvim-neo-tree/neo-tree.nvim/pull/1004

**Pull request merged**